### DISCUSSION TITLE
Xmodem block 128 byte

### DISCUSSION THREAD

**Discussion by Fess1987** [TomNisbet/TommyPROM#62] [General]
Hi Tom! I have assembled your programmer and am trying to program the 28C64 chip via minicom (xmodem). As a result, I get that a code with a size of 30 bytes is supplemented with garbage up to a volume of 128 bytes and written to the EEPROM. From the description of XMODEM, I understood that the protocol transmits data in blocks of 128 bytes. Is it possible to configure the transfer of firmware to the chip via Xmodem with byte-by-byte download? A simple recording of the test firmware (the Zsssss command) records byte-by-byte, that is, not in a block. I'm very confused with this, because furmware, that correctly works in Proteus, doesn't work in real life=/

**Comment by TomNisbet** [comment]
Unfortunately, the XModel protocol only works with blocks, so there is no way to tell it that you are sending a partial block of data.  If you need to just put a small bit of data into the chip, you could use the Poke command.  For example, to put four bytes of data into the chip at address 0xB8, use a command like:
`p 00b8 12 34 56 78`

On Linux or MacOS, the command 'od -t x2' will produce an octal dump that you could cut and paste into the terminal with the Poke command.

**Comment by Fess1987** [comment]
Thanks Tom!

**Comment by Fess1987** [comment]
By the way Is it possible in your TommyProm.ino firmware to make such an algorithm for working through xmodem so that those bytes that follow the data and fill the block up to 128 bytes (garbage) are filled with the value 00h instead of the value that I have now - 01a?

**Reply by TomNisbet** [reply]
In the function XModem::ReceivePacket in Modem.cpp, add this line just before the for loop:
`memset(buffer, 0, bufferSize);`
